src/Integrations/Integrations/SQLSRV/SQLSRVIntegration.php

```diff
@@ -0,0 +1,256 @@
+            $connectionMetadata[Tag::DB_INSTANCE] = $instanceName;
+        }
+
+        $port = '';
+        if (preg_match('/,\s?(\d+)/', $serverName, $matches)) {
+            $port = $matches[1];
+        }
+        $connectionMetadata[Tag::TARGET_PORT] = empty($port) ? '<default>' : $port;
+
+        $connectionInfo = $connectionArguments[1];
+        if (is_array($connectionInfo)) {
+            if (isset($connectionInfo['UID'])) {
+                $connectionMetadata[Tag::DB_USER] = $connectionInfo['UID'];
+            }
+            if (isset($connectionInfo['Database']) && empty($instanceName)) {
+                $connectionMetadata[Tag::DB_INSTANCE] = $connectionInfo['Database'];
+            }
+            if (isset($connectionInfo['CharacterSet'])) {
+                $connectionMetadata['db.charset'] = $connectionInfo['CharacterSet'];
+            }
+        }
+
+        return $connectionMetadata;
+    }
+
+    public static function setDefaultAttributes($source, SpanData $span, $name, $query = null, $result = null)
+    {
+        if (is_array($source)) {
+            $storedConnectionInfo = $source;
+        } else {
+            $storedConnectionInfo = ObjectKVStore::get($source, SQLSRVIntegration::CONNECTION_TAGS_KEY, []);
+        }
+
+        if (!is_array($storedConnectionInfo)) {
+            $storedConnectionInfo = [];
+        }
+
+        $span->name = $name;
+        $span->resource = $query ?? $name;
+        $span->type = Type::SQL;
+        $span->service = 'sqlsrv';
+        $span->meta[Tag::SPAN_KIND] = 'client';
+        $span->meta[Tag::COMPONENT] = SQLSRVIntegration::NAME;
+        $span->meta[Tag::DB_SYSTEM] = SQLSRVIntegration::SYSTEM;
+
+        if ($query) {
+            $span->meta[Tag::DB_STMT] = $query;
+        }
+
+        foreach ($storedConnectionInfo as $tag => $value) {
+            $span->meta[$tag] = $value;
+        }
+    }
+
+    public static function detectError($SQLSRVRetval, SpanData $span)
+    {
+        if ($SQLSRVRetval !== false) {
+            return;
+        }
+
+        $errors = sqlsrv_errors();
+        if (empty($errors)) {
+            return;
+        }
+
+        // There could be multiple errors occurring on the same sqlsrv operation
+        // If this is the case, we concatenate them using ' | ' as the separator
+        // Format: SQL Error: <code>. Driver error: <sqlstate>. Driver-specific error data: <message>
+        $errorMessages = implode(' | ', array_map(function ($error) {
+            return sprintf(
+                'SQL error: %s. Driver error: %s. Driver-specific error data: %s',
+                $error['code'],
+                $error['SQLSTATE'],
+                $error['message']
+            );
+        }, $errors));
+
+        $span->meta[Tag::ERROR_MSG] = $errorMessages;
+        $span->meta[Tag::ERROR_TYPE] = 'SQLSRV error';
+    }
+
+    protected function setMetricNumRows(SpanData $span, $stmt)
+    {
+        if ($stmt) {
```

Comment:
If there's no num_rows, can you please assign sqlsrv_rows_affected() to db.row_count?